src/imitation/algorithms/adversarial/common.py

```diff
@@ -206,22 +204,23 @@ def __init__(
 
         venv = self.venv_buffering = wrappers.BufferingWrapper(self.venv)
 
+        if isinstance(self.reward_train, reward_nets.RewardNet):
```

Comment:
Why is this necessary? I think we can always use `predict_processed` as it's present in base `RewardNet`, just forwards to `predict` by default

Reply:
I think we want to just delete this `isinstance` check? `self.reward_train` type annotation says it's always a `RewardNet` so it should always pass. And the current code does not handle the case where this check does not pass (leaves `reward_fn` undefined).

In fact I think we can probably remove the following line defining `reward_fn` and just inline it into the only place it's used, on line 217.

Reply:
Good point!